core/eolearn/tests/test_core_tasks.py

```diff
@@ -172,47 +173,45 @@ def test_add_rename_remove_feature(patch):
     assert len(patch.mask) == 0, "mask features were not removed"
 
 
-def test_duplicate_feature(patch):
-    mask_data = np.arange(10).reshape(5, 2, 1, 1)
-    feature_name = "MASK1"
-    duplicate_name = "MASK2"
+@pytest.mark.parametrize(
+    "feature_specification",
+    [
+        [(FeatureType.DATA, "bands", "bands2")],
+        [(FeatureType.DATA, "bands", "bands2"), (FeatureType.MASK_TIMELESS, "mask", "mask2")],
+        [(FeatureType.DATA, "bands", f"bands{i}") for i in range(5)],
+    ],
+)
+def test_duplicate_feature(feature_specification: List[FeaturesSpecification], patch: EOPatch) -> None:
+    patch = DuplicateFeatureTask(feature_specification)(patch)
 
-    patch = AddFeatureTask((FeatureType.MASK, feature_name))(patch, mask_data)
+    for f_type, f_name, f_dup_name in feature_specification:
+        assert f_dup_name in patch[f_type]
+        assert id(patch[(f_type, f_name)]) == id(patch[(f_type, f_dup_name)])
+        assert_array_equal(patch[(f_type, f_name)], patch[(f_type, f_dup_name)])
 
-    duplicate_task = DuplicateFeatureTask((FeatureType.MASK, feature_name, duplicate_name))
+
+def test_duplicate_feature_deep(patch: EOPatch) -> None:
+    duplicate_task = DuplicateFeatureTask((FeatureType.DATA, "bands", "bands_dup"))
+    duplicate_task_deep = DuplicateFeatureTask((FeatureType.DATA, "bands", "bands_dup_deep"), deep_copy=True)
     patch = duplicate_task(patch)
+    patch = duplicate_task_deep(patch)
+
+    assert all(f_name in patch.data for f_name in ["bands_dup", "bands_dup_deep"])
+    assert_array_equal(patch.data["bands_dup"], patch.data["bands_dup_deep"])
+    assert id(patch.data["bands_dup"]) != id(patch.data["bands_dup_deep"])
+    patch.data["bands"] += 1
+    assert_array_equal(patch.data["bands_dup"], patch.data["bands"])
+    assert not np.array_equal(patch.data["bands_dup"], patch.data["bands_dup_deep"])
+
 
-    assert duplicate_name in patch.mask, "Feature was not duplicated. Name not found."
-    assert id(patch.mask["MASK1"]) == id(patch.mask["MASK2"])
-    assert np.array_equal(
-        patch.mask[duplicate_name], mask_data
-    ), "Feature was not duplicated correctly. Data does not match."
+def test_duplicate_feature_fails(patch: EOPatch) -> None:
+    duplicate_task = DuplicateFeatureTask((FeatureType.DATA, "bands", "bands_dup"))
+    patch = duplicate_task(patch)
 
     with pytest.raises(ValueError):
         # Expected a ValueError when creating an already exising feature.
         patch = duplicate_task(patch)
```

Comment:
no reason to call it twice, you know what the names in the patch fixture are, use that as input

Reply:
done